pkg/synapse/materialization_test.go

```diff
@@ -0,0 +1,188 @@
+package synapse
+
+import (
+	"testing"
+
+	"github.com/bruin-data/bruin/pkg/pipeline"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMaterializer_Render(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name    string
+		task    *pipeline.Asset
+		query   string
+		want    []string
+		wantErr bool
+	}{
+		{
+			name:  "no materialization, return raw query",
+			task:  &pipeline.Asset{},
+			query: "SELECT 1",
+			want:  []string{"SELECT 1"},
+		},
+		{
+			name: "materialize to a view",
+			task: &pipeline.Asset{
+				Name: "my.asset",
+				Materialization: pipeline.Materialization{
+					Type: pipeline.MaterializationTypeView,
+				},
+			},
+			query: "SELECT 1",
+			want: []string{
+				"^DROP VIEW IF EXISTS my\\.asset;$",
+				"^CREATE VIEW my\\.asset AS SELECT 1;$",
+			},
+		},
+		{
+			name: "materialize to a table, default to create+replace",
+			task: &pipeline.Asset{
+				Name: "my.asset",
+				Materialization: pipeline.Materialization{
+					Type: pipeline.MaterializationTypeTable,
+				},
+			},
+			query: "SELECT 1",
+			want: []string{"^SELECT tmp\\.\\* INTO #bruin_tmp_.+ FROM \\(SELECT 1\\) AS tmp;\n" +
+				"IF OBJECT_ID\\('my\\.asset', 'U'\\) IS NOT NULL DROP TABLE my\\.asset;\n" +
+				"SELECT \\* INTO my\\.asset FROM #bruin_tmp_.+;$"},
```

Comment:
deleting temp table after select into might be useful

Reply:
Missed that. Thanks